tests/backup_restore_test.go

```diff
@@ -245,6 +245,69 @@ func TestPITR(t *testing.T) {
 		err = mysqlRestore.SwapPITRDatabase(ctx, database, timestamp)
 		a.Error(err)
 	})
+
+	t.Run("Schema Migration Failure", func(t *testing.T) {
+		t.Parallel()
+		ctx := context.Background()
+
+		t.Logf("test %s initialize database %s", t.Name(), database)
+
+		mysqlPort := port + 2
+		db, stopFn := initPITRDB(t, localhost, username, database, mysqlPort)
+		defer stopFn()
+		defer db.Close()
+
+		t.Log("insert data")
+		insertRangeData(t, db, 0, numRowsTime0)
+
+		t.Log("make a full backup")
+		driver := getMySQLDriver(ctx, t, localhost, fmt.Sprintf("%d", mysqlPort), username, database)
+		defer func() {
+			err := driver.Close(ctx)
+			a.NoError(err)
+		}()
+
+		buf := doBackup(ctx, t, driver, database)
+		t.Logf("backup content:\n%s\n", buf.String())
+
+		t.Log("insert more data")
+		insertRangeData(t, db, numRowsTime0, numRowsTime1)
+
+		ctxUpdateRow, cancelUpdateRow := context.WithCancel(ctx)
+		t1 := startUpdateRow(ctxUpdateRow, t, username, localhost, database, mysqlPort)
+		t.Logf("start to concurrently update data at t1: %v", t1)
+
+		timestamp := time.Now().Unix()
```

Comment:
nit: change all the `timestamp` to `suffixTs`. It's not the user-provided timestamp for PITR, but a timestamp suffix that will be derived from the `issue.CreatedTs`.